gateway/ln-gateway/src/lnd.rs

```diff
@@ -112,21 +112,30 @@ impl GatewayLndClient {
 
         task_group
             .spawn("LND HTLC Subscription", move |handle| async move {
-                let mut htlc_stream = match client
+                let future_stream = client
                     .router()
-                    .htlc_interceptor(ReceiverStream::new(lnd_rx))
-                    .await
-                    .map_err(|e| LightningRpcError::FailedToGetRouteHints {
-                        failure_reason: format!("Failed to subscribe to LND htlc stream {e:?}"),
-                    }) {
-                    Ok(stream) => stream.into_inner(),
-                    Err(e) => {
-                        error!("Failed to establish htlc stream");
-                        debug!("Error: {:?}", e);
+                    .htlc_interceptor(ReceiverStream::new(lnd_rx));
+                let mut htlc_stream = tokio::select! {
+                    stream = future_stream => {
+                        match stream {
+                            Ok(stream) => stream.into_inner(),
+                            Err(e) => {
+                                error!("Failed to establish htlc stream");
+                                let e = LightningRpcError::FailedToGetRouteHints {
+                                    failure_reason: format!("Failed to subscribe to LND htlc stream {e:?}"),
+                                };
+                                debug!("Error: {e}");
+                                return;
+                            }
+                        }
+                    },
+                    _ = handle.make_shutdown_rx().await => {
```

Comment:
Ugh. This is a little weird that you have to do this. We already wait on the shutdown future below when we select over `message()` and `make_shutdown_rx`. But I think we've seen the `htlc_interceptor` function block before the first HTLC comes in. Is that why this is necessary?